libraries/botbuilder/tests/inspectionMiddleware.test.js

```diff
@@ -291,6 +290,272 @@ describe('InspectionMiddleware', function() {
+            .post('/v3/conversations/Convo1/activities', activity => activity.type === 'trace'
+                && activity.value.text == 'echo: hi')
+            .reply(200, { id: 'test' });
+
+        nock('https://test.com')
+            .post('/v3/conversations/Convo1/activities', activity => activity.type === 'trace'
+                && activity.value.userState && activity.value.userState.x.property == 'hello'
+                && activity.value.conversationState && activity.value.conversationState.y.property == 'world')
+            .reply(200, { id: 'test' });
+
+        // create the various storage and middleware objects we will be using
+
+        const storage = new MemoryStorage();
+        const inspectionState = new InspectionState(storage);
+        const userState = new UserState(storage);
+        const conversationState = new ConversationState(storage);
+        const inspectionMiddleware = new InspectionMiddleware(inspectionState, userState, conversationState);
+
+        // the emulator sends an /INSPECT open command - we can use another adapter here
+
+        const openActivity = MessageFactory.text('/INSPECT open');
+
+        const inspectionAdapter = new TestAdapter(async (turnContext) => {
+            await inspectionMiddleware.processCommand(turnContext);
+        }, null, true);
+
+        await inspectionAdapter.receiveActivity(openActivity);
+
+        const inspectionOpenResultActivity = inspectionAdapter.activityBuffer[0];
+        const attachCommand = inspectionOpenResultActivity.value;
+
+        const adapter = new TestAdapter(async (turnContext) => {
+            try {
+                await inspectionMiddleware.onTurn(turnContext, null)
+                throw new Error('should have thrown an error.')
+            } catch (error) {
+                assert.strictEqual(error.message, 'next is not a function', 'next function should be null')
+            }
+        }, null, true);
+
+        adapter.use(inspectionMiddleware);
+
+        await adapter.receiveActivity(MessageFactory.text(attachCommand));
+
+        await adapter.receiveActivity();
+    });
 
+    it('should invokeInbound throw an error', async () => {
+        let warnExecuted = false;
+        const storage = new MemoryStorage();
+        const inspectionState = new InspectionState(storage);
+        const userState = new UserState(storage);
+        const conversationState = new ConversationState(storage);
+        const inspectionMiddleware = new InspectionMiddleware(inspectionState, userState, conversationState);
+
+        const warn = console.warn;
+
+        console.warn = (message) => {
+            const condition = message.startsWith('Exception in inbound interception')
+            assert(condition, 'should have throw an error')
+            warnExecuted = true;
+        }
+
+        await inspectionMiddleware.invokeInbound();
+
+        if(!warnExecuted) {
+            throw new Error('invokeInbound should have throw an error')
```

Comment:
Is there an advantage to throwing an error here, vs assert(warnExecuted, '...);?